### PR TITLE
chore: upgraded to v2.1.0-rc1

### PR DESCRIPTION
**Pull request by Orland0x**
got a nice LOC reduction from `Store` derives

### CODE REVIEW

File: starknet/src/utils/types.cairo [hunk 1]
```diff
@@ -37,7 +36,7 @@ enum Choice {
     Abstain: (),
 }
 
-#[derive(Copy, Drop, Serde, PartialEq)]
+#[derive(Drop, Serde, PartialEq, Copy, starknet::Store)]
```
Comment (pscott) [inline starknet/src/utils/types.cairo:39]: why `starknet::Store` and not `Store` ?

File: starknet/src/utils/types.cairo [hunk 2]
```diff
@@ -438,74 +215,8 @@ impl StorageAccessFelt252Array of StorageAccess<Array<felt252>> {
...
-            offset,
-            value.params
-        )
-    }
-
-    fn size_internal(value: Strategy) -> u8 {
-        // Add 1 for the strategy address
-        StorageAccess::size_internal(value.params) + 1
+    fn size() -> u8 {
+        255_u8
```
Comment (pscott) [inline starknet/src/utils/types.cairo:220]: why 255?

File: starknet/src/utils/types.cairo (hunk 1 above)
Comment (Orland0x) [reply starknet/src/utils/types.cairo:39]: for some reason it wouldnt compile with `Store`

Comment (Orland0x) [reply starknet/src/utils/types.cairo:39]: not sure why

File: starknet/src/utils/types.cairo (hunk 2 above)
Comment (Orland0x) [reply starknet/src/utils/types.cairo:220]: now size() doesnt take an argument so there is no way to know the size of dynamic types. I spoke to eni about this today and we agreed it made the most sense to put the max size possible. 

Ill add a comment to clarify

Comment (pscott) [reply starknet/src/utils/types.cairo:220]: ugh yeah this needs to be take a snapshot... gotcha

File: starknet/src/utils/types.cairo (hunk 1 above)
Comment (pscott) [reply starknet/src/utils/types.cairo:39]: weird... I don't see why it wouldn't but ok !

**Review by pscott — APPROVED** [review]
utACK

**Pull request merged**